File: app/src/main/java/com/nathaniel/motus/cavevin/view/EditDialogFragment.java
package com.nathaniel.motus.cavevin.view;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;

import com.nathaniel.motus.cavevin.R;
import com.nathaniel.motus.cavevin.controller.CellarInputUtils;

/**
 * A simple {@link Fragment} subclass.
 */
public class EditDialogFragment extends DialogFragment {
    //abstract class that shows a dialog with an EditText

    private EditText mInputTextEdit;
    private TextView mInviteText;
    private onEditDialogClickListener mCallBack;
    private String mDialogTitle;
    private String mInvite;
    private String mPreFilledInput;

    public interface onEditDialogClickListener{
//        Interface needed for callback

        public void onEditDialogClick(View v, String inputText);
    }

    public EditDialogFragment() {
        // Required empty public constructor
    }

    public static EditDialogFragment newInstance(String dialogTitle, String invite, String preFilledInput){
//        Create a new instance

        EditDialogFragment edf=new EditDialogFragment();
        edf.mDialogTitle=dialogTitle;
        edf.mInvite=invite;
        edf.mPreFilledInput=preFilledInput;

        return edf;
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder adb=new AlertDialog.Builder(getActivity());
        final LayoutInflater inflater=getActivity().getLayoutInflater();

        final View view=inflater.inflate(R.layout.fragment_edit_dialog,null);
        mInputTextEdit=view.findViewById(R.id.fragment_edit_dialog_edit);
        mInviteText=view.findViewById(R.id.fragment_edit_dialog_text);

        mInputTextEdit.setText(mPreFilledInput);
        mInviteText.setText(mInvite);

        adb.setView(view);
        adb.setTitle(mDialogTitle);
        adb.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                createCallBackToParentActivity();
                String inputText= CellarInputUtils.replaceForbiddenCharacters(view.getContext(),mInputTextEdit.getText().toString());
                mCallBack.onEditDialogClick(view,inputText);
            }
        });
        adb.setNegativeButton("CANCEL", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                if (dialog!=null)
                    dialog.dismiss();
            }
        });

        return adb.create();
    }

    private void createCallBackToParentActivity(){

        mCallBack=(onEditDialogClickListener)getActivity();
    }

}

File: app/src/main/java/com/nathaniel/motus/cavevin/view/EditDialogFragment.kt
package com.nathaniel.motus.cavevin.view

import com.nathaniel.motus.cavevin.controller.CellarInputUtils.replaceForbiddenCharacters
import android.widget.TextView
import com.nathaniel.motus.cavevin.R
import android.os.Bundle
import android.widget.EditText
import android.app.AlertDialog
import android.app.Dialog
import android.view.View
import com.nathaniel.motus.cavevin.R.layout
import androidx.fragment.app.DialogFragment

/**
 * A simple [Fragment] subclass.
 */
class EditDialogFragment : DialogFragment() {
    //abstract class that shows a dialog with an EditText
    private lateinit var mInputTextEdit: EditText
    private lateinit var mInviteText: TextView
    private var mCallBack: onEditDialogClickListener? = null
    private var mDialogTitle: String? = null
    private var mInvite: String? = null
    private var mPreFilledInput: String? = null

    interface onEditDialogClickListener {
        //        Interface needed for callback
        fun onEditDialogClick(v: View?, inputText: String?)
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val adb = AlertDialog.Builder(activity)
        val inflater = activity!!.layoutInflater
        val view = inflater.inflate(layout.fragment_edit_dialog, null)
        mInputTextEdit = view.findViewById(R.id.fragment_edit_dialog_edit)
        mInviteText = view.findViewById(R.id.fragment_edit_dialog_text)
        mInputTextEdit.setText(mPreFilledInput)
        mInviteText.text = mInvite
        adb.setView(view)
        adb.setTitle(mDialogTitle)
        adb.setPositiveButton("OK") { dialog, which ->
            createCallBackToParentActivity()
            val inputText =
                replaceForbiddenCharacters(view.context, mInputTextEdit.getText().toString())
            mCallBack!!.onEditDialogClick(view, inputText)
        }
        adb.setNegativeButton("CANCEL") { dialog, which -> dialog?.dismiss() }
        return adb.create()
    }

    private fun createCallBackToParentActivity() {
        mCallBack = activity as onEditDialogClickListener?
    }

    companion object {
        fun newInstance(
            dialogTitle: String?,
            invite: String?,
            preFilledInput: String?
        ): EditDialogFragment {
//        Create a new instance
            val edf = EditDialogFragment()
            edf.mDialogTitle = dialogTitle
            edf.mInvite = invite
            edf.mPreFilledInput = preFilledInput
            return edf
        }
    }
}